coalib/settings/ConfigurationGathering.py

```diff
@@ -109,11 +109,18 @@ def warn_nonexistent_targets(targets, sections, log_printer):
     :param sections:    The sections to search. (Dict.)
     :param log_printer: The log printer to warn to.
     """
+
+    sections_substrings = {
+        substring
+        for sections_name in sections
+        for substring in sections_name.split('.')
+    }
     for target in targets:
-        if target not in sections:
+        if target not in sections_substrings | set(sections):
```

Comment:
if I have sections

```
[all.python]
[all.python.style]
[other.python]
```

and I run `coala all.python` according to this new logic I'd expect the first two sections to be run right?

Maybe something like:

```

sections_name = ['.' + section + '.' for section in sections_name]
for target in targets:
    target = '.' + target + '.'
    if any(target in secname for secname in sections_name):
       ...
```